.github/workflows/lego.yml

```diff
@@ -0,0 +1,142 @@
+name: LEGO frontend
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!master'
+      - '!prod'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout commit
+      uses: actions/checkout@v1
+
+    - name: Set up Node 11.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '11.x'
+
+    - name: Install dependencies
+      run: yarn --forzen-lockfile
+
+    - name: Build server
+      run: yarn build:server
+
+    - name: Build client
+      run: yarn build:client
+
+
+  tests:
+    name: Run tests
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout commit
+      uses: actions/checkout@v1
+
+    - name: Set up Node 11.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '11.x'
+
+    - name: Install dependencies
+      run: yarn --forzen-lockfile
+
+    - name: Run tests
+      run: yarn test
+
+  linting:
+    name: Run linting
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout commit
+      uses: actions/checkout@v1
+
+    - name: Set up Node 11.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '11.x'
+
+    - name: Install dependencies
+      run: yarn --forzen-lockfile
+
+    - name: Run linting
+      run: yarn lint
+
+  flow:
+    name: Run flow
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout commit
+      uses: actions/checkout@v1
+
+    - name: Set up Node 11.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '11.x'
+
+    - name: Install dependencies
+      run: yarn --forzen-lockfile
+
+    - name: Run flow
+      run: yarn flow
+
+  cypress:
+    name: Run Cypress
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:9.5
+        ports:
+        - 5432:5432
+        env:
+          POSTGRES_USER: lego
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      thumbor:
+        image: apsl/thumbor:latest
+        ports:
+        - 8001:8000
+        env:
+          SECURITY_KEY: lego-dev
+          MAX_WIDTH: 1000
+          MAX_HEIGHT: 800
+          QUALITY: 98
+          ALLOW_UNSAFE_URL: 0
+          ALLOW_OLD_URLS: 0
+          AWS_ACCESS_KEY_ID: lego-dev
+          AWS_SECRET_ACCESS_KEY: lego-dev
+          TC_AWS_LOADER_BUCKET: lego
+          TC_AWS_REGION: us-east-1
+          TC_AWS_ENDPOINT: http://localhost:9000
+          LOADER: tc_aws.loaders.s3_loader
+      redis:
+        image: redis
+        ports:
+        - 6379:6379
+
+    steps:
+    - name: Checkout commit
+      uses: actions/checkout@v1
+
+    - name: Set up Node 11.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '11.x'
+
+    - name: Install dependencies
+      run: yarn --forzen-lockfile
+
+    - name: Setup MinIO
```

Comment:
Can't minio run as a service like postgres and redis?

Reply:
Looks like it was for debugging purposes. Should be ok to move it into a service now.

Is it possible to look at the logs for the services? :thinking: Can't find a way to do it right now. :/

Reply:
I'm trying different solutions. I need to run certain commands within the container, and it's not possible to send/run commands there. That's why you have to create an Action for that specific service (or a new image that run the commands).

In addition, there are no caching options (i.e. for node_modules) currently. The only options is to upload and download artifacts, but I tested this and it was too slow for node_modules (50k+ files!). Cache is supposed to be implemented in mid-November, https://github.community/t5/GitHub-Actions/Caching-files-between-GitHub-Action-executions/m-p/30974/highlight/true#M630.

I am still testing this and different ways to implement Actions, the documentation isn't always that detailed.

Regarding logs for services, I haven't found a way to look at them yet... Will test some more when I get some time.